engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -271,20 +261,19 @@ private void preRenderUpdate(RenderingStage renderingStage) {
         resetStats();
 
         currentRenderingStage = renderingStage;
-        if (currentRenderingStage == RenderingStage.MONO || currentRenderingStage == RenderingStage.LEFT_EYE) {
-            isFirstRenderingStageForCurrentFrame = true;
-        } else {
-            isFirstRenderingStageForCurrentFrame = false;
-        }
+        isFirstRenderingStageForCurrentFrame = (currentRenderingStage == RenderingStage.MONO) || (currentRenderingStage == RenderingStage.LEFT_EYE);
 
         // this is done to execute this code block only once per frame
         // instead of once per eye in a stereo setup.
         if (isFirstRenderingStageForCurrentFrame) {
             timeSmoothedMainLightIntensity = TeraMath.lerp(timeSmoothedMainLightIntensity, getMainLightIntensityAt(playerCamera.getPosition()), secondsSinceLastFrame);
 
             playerCamera.update(secondsSinceLastFrame);
-            positionShadowMapCamera();
-            shadowMapCamera.update(secondsSinceLastFrame);
+
+            for (Node node : renderingPipeline) {
+                node.update(secondsSinceLastFrame);
+            }
+            
```

Comment:
I think `update()` is needed for updating shadowMapCamera (for only shadowMap node) and setting FBOs to nodes. Therefore, I think removing `update()` method would break existing functionality.